tests/cpp/test_host_irs.cpp

```diff
@@ -387,6 +389,145 @@ INSTANTIATE_TEST_SUITE_P(
+  auto set_stream = IrBuilder::create<PostOnStream>(
+      static_cast<IrContainer*>(hic.get()),
+      stream,
+      std::vector<Val*>(),
+      std::vector<Val*>());
+  hic->pushBackTopLevelExprs(set_stream);
+
+  HostIrExecutor hie(std::move(hic));
+  setCurrentCUDAStream(c10::cuda::getDefaultCUDAStream(0));
+  hie.runWithInput({});
+  EXPECT_NE(
+      c10::cuda::getDefaultCUDAStream(0), c10::cuda::getCurrentCUDAStream(0));
+}
+
+using StreamHostIrTestParams = std::tuple<bool, int, int>;
+using StreamHostIrTest = NVFuserFixtureParamTest<StreamHostIrTestParams>;
+
+// The following test execute the same fusion `n_iterations` times by posting
+// the kernels on `n_streams` different streams in a Round-Robin fashion. We
+// thus produce `n_iterations` outputs from the same input, with a potential
+// overlap of n_streams/n_iterations
+TEST_P(StreamHostIrTest, SingleFusionMultipleStreams) {
+  auto [use_fusion_executor_cache, n_streams, n_iterations] = GetParam();
+
+  // [Step 1)] Define the Fusion we want to execute
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  std::vector<int64_t> input_sizes = {4, 8, 32};
+
+  auto tv0 = makeConcreteTensor(input_sizes);
+  auto tv1 = add(tv0, tv0);
+  auto tv2 = sum(tv1, {0});
+  fusion->addInput(tv0);
+  fusion->addOutput(tv2);
+
+  // [Step 2)] Instantiate an HostIroCntainer
+  auto hic = std::make_unique<HostIrContainer>();
+  FusionGuard::setCurFusion(hic.get());
+
+  // Create N different Streams
+  std::vector<Stream*> streams;
+  for (int i = 0; i < n_streams; i++) {
+    streams.push_back(
+        IrBuilder::create<Stream>(static_cast<IrContainer*>(hic.get())));
+  }
+
+  // [Step 3)] Create a HostUnit Ir holding the created fusion
+  auto host_unit = IrBuilder::create<HostUnit>(
+      static_cast<IrContainer*>(hic.get()), std::move(fusion));
+
+  // [Step 4)] Create TensorViews representing the Fusion's inputs at the Host
+  // level
+  IrCloner ir_cloner_input(hic.get());
+  std::vector<Val*> post_on_stream_inputs = {
+      ir_cloner_input.clone(host_unit->fusion_to_execute()->inputs().at(0))};
+  hic->addInput(post_on_stream_inputs.at(0));
+
+  for (int i = 0; i < n_iterations; i++) {
+    // [Step 4)] Create TensorViews representing the Fusion's ouputs at the Host
+    // level
+    IrCloner ir_cloner_output(hic.get());
+    std::vector<Val*> post_on_stream_outputs = {ir_cloner_output.clone(
+        host_unit->fusion_to_execute()->outputs().at(0))};
+
+    // [Step 5)] Create a PostOnStream Ir representing executing the Fusion with
+    // given I/O
+    auto post_on_stream = IrBuilder::create<PostOnStream>(
+        static_cast<IrContainer*>(hic.get()),
+        host_unit,
+        post_on_stream_inputs,
+        post_on_stream_outputs);
+
+    // Set the Stream
+    auto set_stream = IrBuilder::create<PostOnStream>(
+        static_cast<IrContainer*>(hic.get()),
+        streams[i % streams.size()],
+        std::vector<Val*>(),
```

Comment:
anticipating a possible comment: the compiler complains if we only write `{}` here:

```
/opt/pytorch/Fuser/tests/cpp/test_host_irs.cpp:397:52: error: no matching function for call to ‘nvfuser::IrBuilder::create<nvfuser::
hir::PostOnStream>(nvfuser::IrContainer*, nvfuser::hir::Stream*&, <brace-enclosed initializer list>, <brace-enclosed initializer lis
t>)’                                                                                                                                  397 |   auto set_stream = IrBuilder::create<PostOnStream>(
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  398 |       static_cast<IrContainer*>(hic.get()),
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  399 |       stream,                                                                                                                     |       ~~~~~~~
  400 |       {},
      |       ~~~
  401 |       {});
      |       ~~~
In file included from /opt/pytorch/Fuser/csrc/host_ir/host_ir.h:12,
                 from /opt/pytorch/Fuser/csrc/ir/all_nodes.h:10,
                 from /opt/pytorch/Fuser/csrc/device_lower/analysis/trivial_broadcast.h:11,
                 from /opt/pytorch/Fuser/csrc/compute_at_map.h:10,
                 from /opt/pytorch/Fuser/csrc/device_lower/lower2device.h:12,
                 from /opt/pytorch/Fuser/csrc/evaluator_common.h:9,
                 from /opt/pytorch/Fuser/csrc/expr_evaluator.h:10,
                 from /opt/pytorch/Fuser/csrc/executor_kernel_arg.h:12,
                 from /opt/pytorch/Fuser/tests/cpp/test_host_irs.cpp:10:
/opt/pytorch/Fuser/csrc/ir/builder.h:30:13: note: candidate: ‘static T* nvfuser::IrBuilder::create(Args&& ...) [with T = nvfuser::hi
r::PostOnStream; Args = {}]’                                                                                                           30 |   static T* create(Args&&... args) {
      |             ^~~~~~
/opt/pytorch/Fuser/csrc/ir/builder.h:30:13: note:   candidate expects 0 arguments, 4 provided
/opt/pytorch/Fuser/csrc/ir/builder.h:44:13: note: candidate: ‘static T* nvfuser::IrBuilder::create(nvfuser::IrContainer*, Args&& ...
) [with T = nvfuser::hir::PostOnStream; Args = {}]’
   44 |   static T* create(IrContainer* container, Args&&... args) {
      |             ^~~~~~
/opt/pytorch/Fuser/csrc/ir/builder.h:44:13: note:   candidate expects 1 argument, 4 provided
```

Reply:
I think https://github.com/NVIDIA/Fuser/pull/2337 may help with this. Worth a quick try.

Reply:
done!